website/content/middleware/timeout.md

```diff
@@ -8,48 +8,62 @@ description = "Timeout middleware for Echo"
 
 Timeout middleware is used to timeout at a long running operation within a predefined period.
 
+When timeout occurs, and the client receives timeout response the handler keeps running its code and keeps using resources until it finishes and returns!
```

Comment:
Sorry for posting a comment too late. When I read this the first time I was not sure if it's true, so after a while playing around with the Timeout middleware a found a bug in it. Please see https://github.com/labstack/echo/issues/1909
So, I think this should say that the handler must handle the Request Context properly in order to finish when the middleware timeouts.

Reply:
no, only "reasonable" usecase for timeout middleware is to let handler execute until it normally finishes/exists regardless what happens with coroutine serving request. If this usecase is not needed - then timeout middleware is not needed. you could just create new context with timeout and be fine.

Reply:
"reasonable" i.e.  if client disconnects prematurely the db transaction would still get finished and commited.

if we start cancelling contexts it would mean that `conn.ExecContext(ctx, query, args)` would get cancelled and work would not get commited.

Reply:
probably should indicate that also in example that `doBusinessLogic` takes in will get deadlined.

Reply:
I think `timeout` has 2 properties that user may seek/need:

* to get response before long-running handler actually finishes  (for them even getting http.StatusServiceUnavailable better than nothing)
* to let handler finish work regardless what happens to the client (for them getting unfinished work commited is essential)

Reply:
Well, for me the timeout middleware was to ensure that your handler do the work you want in a particular time period, and if that isn't true you get a response that indicates that. And for that reason all the work should be stop. But your descriptions are more like a middleware to do async work, and of course you are right, if that is the case the work must continue until it is finished. Maybe we need to decide which of this two behaviors will be supported by this middleware an rename it to clarify our intention if it is needed.